lib/marisa/grimoire/vector/vector.h

```diff
@@ -202,24 +194,30 @@ class Vector {
   }
   void swap(Vector &rhs) {
     buf_.swap(rhs.buf_);
-    marisa::swap(objs_, rhs.objs_);
     marisa::swap(const_objs_, rhs.const_objs_);
     marisa::swap(size_, rhs.size_);
     marisa::swap(capacity_, rhs.capacity_);
-    marisa::swap(fixed_, rhs.fixed_);
   }
 
   static std::size_t max_size() {
     return MARISA_SIZE_MAX / sizeof(T);
   }
 
  private:
-  std::unique_ptr<char[]> buf_;
-  T *objs_ = nullptr;
+  struct AlignedArrayDeleter {
+    void operator()(char *p) const {
+      operator delete[](p, std::align_val_t{alignof(T)});
+    };
+  };
+
+  std::unique_ptr<char[], AlignedArrayDeleter> buf_;
   const T *const_objs_ = nullptr;
   std::size_t size_ = 0;
   std::size_t capacity_ = 0;
-  bool fixed_ = false;
+
+  T *objs() {
+    return std::launder(reinterpret_cast<T *>(buf_.get()));
```

Comment:
I saw you used to have `assume_aligned`, which is C++20.
Is there a TODO: Use assume_aligned when C++20 is ok? 

There's also `__builtin_assume_aligned`

Reply:
Replied below